javatests/arcs/core/analysis/ExpressionClaimDeducerTest.kt

```diff
@@ -0,0 +1,338 @@
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x", "foo"),
+        listOf("y", "foo", "bar")
+      )
+    )
+  }
+
+  @Test
+  fun new_field_access_binexpr() {
+    val expr = PaxelParser.parse("new Object {foo: input.foo, bar: input.foo.bar + input.foo}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("input", "foo"),
+        "bar" to Deduction.Paths(
+          listOf("input", "foo", "bar"),
+          listOf("input", "foo")
+        )
+      )
+    )
+  }
+
+  @Test
+  fun field_access_nested_expr() {
+    val expr = PaxelParser.parse("(new Foo {x: foo, y: bar}).x")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "x" to Deduction.Path("foo")
+      )
+    )
+  }
+
+  @Test
+  fun field_access_nested_multi_field_expr() {
+    val expr = PaxelParser.parse("(new Foo {a: foo.x + foo.y, b: foo.y + foo.z}).a")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "a" to Deduction.Paths(
+          listOf("foo", "x"),
+          listOf("foo", "y")
+        )
+      )
+    )
+  }
+
+  @Test
+  fun new_nested() {
+    val expr = PaxelParser.parse("new Foo {a: (new Bar {x: cat, y: dog}), b: foo, c: 5}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "a" to Deduction.Scope(
+          "x" to Deduction.Path("cat"),
+          "y" to Deduction.Path("dog")
+        ),
+        "b" to Deduction.Path("foo"),
+        "c" to Deduction.Path()
+      )
+    )
+  }
+
+  @Test
+  fun new_deeply_nested() {
+    val expr = PaxelParser.parse(
+      "new Foo {x: (new Bar {y: (new Baz {z: (new Buz {a: foo}) }) }) }"
+    )
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "x" to Deduction.Scope(
+          "y" to Deduction.Scope(
+            "z" to Deduction.Scope(
+              "a" to Deduction.Path("foo")
+            )
+          )
+        )
+      )
+    )
+  }
+
+  @Test
+  fun from_select() {
+    val expr = PaxelParser.parse("from f in foo select f")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        Deduction.Scope(
+          "f" to Deduction.Path("foo")
+        ),
+        Deduction.Equal(
+          Deduction.Path("foo")
+        )
+      )
+    )
+  }
+
+  @Test
+  fun from_select_field() {
+    val expr = PaxelParser.parse("from f in foo.input.baz select f.x.bar")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        Deduction.Scope(
```

Comment:
Should I introduce a new `Deduction` case class for Aliases?

Reply:
This way, they can be ignored in the conversion to Claims + AccessPaths.

Reply:
Well, I don't know what you would like to to with this `Deduction` case.

I would think that if we follow your structure we would be getting:

For `from f in foo.input.baz`
We would get:
```
Deduction.Scope(
  "f" to Deduction.Path("foo", "input", "baz")
)
```

For `from f in foo.input.baz select f.x.bar`
We would get:
```
Deduction.Equal(
  Deduction.Path("foo", "input", "baz", "x", "bar")
)
```

I would imagine that the way this would work is that you analyze the expression in select (here: `f.x.bar`) and perform the substitutions from the scope that you received from analyzing the qualifier of the select (here: `from f in foo.input.baz`).

In other words, no, I don't see a need for another thing for Aliases.